### PR TITLE
Specializing static and media based on tenants

### PR DESCRIPTION
**Pull request by mateuspadua**
New Feature to be possible have **different static files for each tenant or upload media files to separate directories for each tenant**. More information, please read the documentation on this commit. :)

### CODE REVIEW

**Comment by kosz85** [comment]
Nice addition, but... From my personal story :) Using `schema_name` in `path` can be a bit problematic, especially when you can (as app feature) or have to (as client request) rename `schema_name`. I have less advance solution of this (just constructing path where it's needed), but really using `tenant.id` is much better option, as even after renaming it stays the same. It could be an option if you want to use `schema_name` or `id`, it's small change but it will save you problems in future.

**Comment by mateuspadua** [comment]
Thank you for your considerations. I followed the same idea of what already existed for the templates based on tenants, so I used schema_name, but later, we can put a flag in the settings where we can choose between using ID or schema_name. :). Thanks

File: django_tenants/middleware.py [hunk 1]
```diff
@@ -26,7 +32,9 @@ def process_request(self, request):
         domain = get_object_or_404(get_tenant_domain_model().objects.select_related('tenant'),
                                    domain=hostname)
 
+        domain.tenant.domain_url = hostname
```
Comment (tomturner) [inline django_tenants/middleware.py]: @mateuspadua Is this line needed as i don't think it is

Comment (mateuspadua) [inline django_tenants/middleware.py]: I dont add the line 35, this line already existed, see here. :)
https://github.com/tomturner/django-tenants/blob/master/django_tenants/middleware.py

**Comment by mateuspadua** [comment]
Hi @tomturner, i'm here if you need any help to merge my pull request. Thanks

**Comment by tomturner** [comment]
@mateuspadua Thank you I will try and merge over the weekend

**Comment by mateuspadua** [comment]
@tomturner thanks for merge, i believe that it is going to help many people. :)

**Pull request merged**